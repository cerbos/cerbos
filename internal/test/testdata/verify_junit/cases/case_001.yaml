---
description: Valid test suite with some failing tests