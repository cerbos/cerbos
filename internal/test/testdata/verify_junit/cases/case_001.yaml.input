-- suite_test.yaml --
---
name: CompanyTestSuite
description: Tests for verifying the company resource policy
principals:
  admin:
    id: admin
    roles:
      - admin
  user:
    id: user
    roles:
      - user
resources:
  company:
    id: xx1
    kind: company

tests:
  - name: Company Test 1
    input:
      principals:
        - admin
      resources:
        - company
      actions:
        - create
        - read
    expected:
      - principal: admin
        resource: company
        actions:
          create: EFFECT_DENY
          read: EFFECT_DENY

  - name: Company Test 2
    input:
      principals:
        - user
      resources:
        - company
      actions:
        - create
        - read
    expected:
      - principal: user
        resource: company
        actions:
          create: EFFECT_DENY
          read: EFFECT_DENY
