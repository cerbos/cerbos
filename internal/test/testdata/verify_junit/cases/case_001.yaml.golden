<testsuites errors="0" failures="4" skipped="0" tests="4">
  <testsuite description="Tests for verifying the company resource policy" name="CompanyTestSuite" file="suite_test.yaml" errors="0" failures="4" skipped="0" tests="4">
    <properties></properties>
    <testcase file="suite_test.yaml" classname="admin.company.create" name="Company Test 1">
      <failure type="RESULT_FAILED">
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_DENY</expected>
      </failure>
      <properties></properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="admin.company.read" name="Company Test 1">
      <failure type="RESULT_FAILED">
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_DENY</expected>
      </failure>
      <properties></properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="user.company.create" name="Company Test 2">
      <failure type="RESULT_FAILED">
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_DENY</expected>
      </failure>
      <properties></properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="user.company.read" name="Company Test 2">
      <failure type="RESULT_FAILED">
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_DENY</expected>
      </failure>
      <properties></properties>
    </testcase>
  </testsuite>
</testsuites>